src/preferences.ts

```diff
@@ -15,66 +16,84 @@ export const updatePreference = (section: prefSection, value: any) => {
         .update(section, value);
 };
 
-export const getSaveLocationPref = (): string => getPreference('saveLocation');
+export const getSaveLocationPref = (): string =>
+    getPreference('general.saveLocation');
 
-export const getTimeOutPref = (): number => getPreference('timeOut') || 3000;
+export const getTimeOutPref = (): number =>
+    getPreference('general.timeOut') || 3000;
 
 export const getCppArgsPref = (): string[] =>
-    getPreference('argsCpp').split(' ') || [];
+    getPreference('language.cppArgs').split(' ') || [];
 
 export const getCArgsPref = (): string[] =>
-    getPreference('argsC').split(' ') || [];
+    getPreference('language.cArgs').split(' ') || [];
 
 export const getPythonArgsPref = (): string[] =>
-    getPreference('argsPython').split(' ') || [];
+    getPreference('language.pythonArgs').split(' ') || [];
 
 export const getRustArgsPref = (): string[] =>
-    getPreference('argsRust').split(' ') || [];
+    getPreference('language.rustArgs').split(' ') || [];
 
 export const getJavaArgsPref = (): string[] =>
-    getPreference('argsJava').split(' ') || [];
+    getPreference('language.javaArgs').split(' ') || [];
 
 export const getFirstTimePref = (): boolean =>
-    getPreference('firstTime') || 'true';
+    getPreference('general.firstTime') || 'true';
 
 export const getDefaultLangPref = (): string | null => {
-    const pref = getPreference('defaultLanguage');
+    const pref = getPreference('general.defaultLanguage');
     if (pref === 'none' || pref == ' ' || !pref) {
         return null;
     }
     return pref;
 };
 
 export const useShortCodeForcesName = (): boolean => {
-    return getPreference('useShortCodeforcesName');
+    return getPreference('general.useShortCodeforcesName');
 };
 
+export const getPythonCommand = (): string =>
+    getPreference('language.pythonCommand') || 'python';
+
+export const getMenuChoices = (): string[] =>
+    getPreference('general.menuChoices').split(' ') || [];
+
 /** The language ID preference for a particular extension. */
 export const getLanguageId = (srcPath: string): number => {
```

Comment:
Changes in this **function** : Instead of getting compiler ids, get the compiler name from respective setting and then use the map `compilerToId` to get the respective id.